code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/InputBoxNotificationBuilderTest.kt

```diff
@@ -0,0 +1,180 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.InputBoxPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MOCKED_HINT
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockInputBoxPushTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedInputBoxPushTemplateWithRequiredData
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.MapData
+import junit.framework.TestCase
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.Shadows
+import org.robolectric.annotation.Config
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class InputBoxNotificationBuilderTest {
+    @Mock
+    private lateinit var context: Context
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        context = RuntimeEnvironment.getApplication()
+        trackerActivityClass = DummyActivity::class.java
+        broadcastReceiverClass = DummyBroadcastReceiver::class.java
+    }
+
+    @Test
+    fun `construct should return a NotificationCompat Builder`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        TestCase.assertEquals(NotificationCompat.Builder::class.java, notificationBuilder.javaClass)
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test
+    fun `construct should not have any inputText action if the template is created from intent`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData(true)
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertEquals(0, notificationBuilder.mActions.size)
+    }
+
+    @Test
+    fun `construct should have an InputText action if the template is not created from intent`() {
+        val pushTemplate = provideMockedInputBoxPushTemplateWithRequiredData()
+        val notificationBuilder = InputBoxNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertEquals(1, notificationBuilder.mActions.size)
+        assertNotNull(notificationBuilder)
```

Comment:
can we reverse the order of the assertions? e.g., have `assertNotNull(notificationBuilder)` first then `assertEquals(1, notificationBuilder.mActions.size)`